modin/engines/base/frame/data.py

```diff
@@ -1674,7 +1706,9 @@ def _copartition(self, axis, other, how, sort, force_repartition=False):
         reindexed_other_list = []
 
         for i in range(len(other)):
-            if right_old_idxes[i].equals(joined_index) and not force_repartition:
+            if (
+                right_old_idxes[i].equals(joined_index) and not force_repartition
+            ) or is_aligning_applied:
```

Comment:
we can slightly optimize this, by changing the order of components in `if` statement:
```python
if is_aligning_applied or (
    right_old_idxes[i].equals(joined_index) and not force_repartition
    ):
```
in that case, if aligning applied then we don't need to trigger checking axis equality, which could be slow in case of long axis

Reply:
fixed